container/container_contract.go

```diff
@@ -182,6 +205,31 @@ func Put(container []byte, signature interop.Signature, publicKey interop.Public
 
 	addContainer(ctx, containerID, ownerID, cnr)
 
+	if name != "" {
+		if zone == "" {
+			zone = storage.Get(ctx, nnsRootKey).(string)
+		}
+		nnsContractAddr := storage.Get(ctx, nnsContractKey).(interop.Hash160)
+		domain := name + "." + zone
+		isAvail := contract.Call(nnsContractAddr, "isAvailable", contract.ReadStates|contract.AllowCall, domain).(bool)
+		if isAvail {
+			res := contract.Call(nnsContractAddr, "register", contract.All, domain, runtime.GetExecutingScriptHash()).(bool)
+			if !res {
+				panic("can't register the domain " + domain)
+			}
+		} else {
+			res := contract.Call(nnsContractAddr, "getRecord", contract.ReadStates|contract.AllowCall, domain, 16 /* TXT */)
+			if res != nil {
+				panic("name is already taken")
+			}
+		}
```

Comment:
Notary disabled environment will benefit if sanity check will happen before notification. In this case, test invocation will fail before Storage node will return response to the client. Therefore client will get nice error "container name is already taken").

For example we have balance check up there for this reason:
```go
	if !alphabetCall {
		if balance < containerFee*len(alphabet) {
			panic("insufficient balance to create container")
		}
		runtime.Notify("containerPut", container, signature, publicKey, token)
		return
	}
```

Reply:
What's the reason for this behaviour? I thought failed execution should have no effect on subscribers.

Reply:
> I thought failed transactions should have no effect on subscribers.

Indeed, they have no effect on subscribers. The reason for this behavior -- _at notary disabled environment_ return error to the client as soon as possible for better UX. 

Compare these two cases:

1. Domain name sanity check at container creation:
![image](https://user-images.githubusercontent.com/6323066/135602244-014fbc0f-63c2-4de1-8213-c31e02a93294.png)


2. Domain name sanity check before `containerPut` notification.
![image](https://user-images.githubusercontent.com/6323066/135602716-d22bf53f-99cd-433a-8204-7cbce7abc380.png)